### PR TITLE
[WIP] Setup Search with algolia

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Set up Algolia doc search for seneca. cf #127
https://github.com/algolia/docsearch

This is not complete yet, stying as yet to be done to match website.

I think we should go for something like eslint: 
cf http://eslint.org/

> [WOW ! Much search Such #doc SO #DX AMAAAZE](https://twitter.com/AdrienBecchis/status/688253946020823041)

### CODE REVIEW

**Comment by pixelastic** [comment]
Just to give an idea what the default styling looks like:
image

But you will find in [our documentation](https://community.algolia.com/docsearch/documentation/) everything you need to adapt the colors to your website.

**Comment by AdrieanKhisbe** [comment]
Thanks tim!! 

(I started to hack around the color, cf last comit I had forgot to push: 10df323.
But Didn't seems to work. (but might be from scss and co). I'll dig later

**Comment by pixelastic** [comment]
You have two ways to update the colors. The easiest is to copy paste the CSS example from our doc, update the colors, and add that to your own CSS files.

The other is to update the `_variables.scss` file in our repo and run `npm run build:css`. This will regenerate the whole CSS file. You'll then just have to remove the link to the CSS file from the CDN and replace it with your newly generated one.

Which solution you choose is up to you and your build system :)

**Comment by AdrieanKhisbe** [comment]
thanks for these details! :)

**Comment by mcdonnelldean** [comment]
@AdrieanKhisbe Great job organising this, very happy with it! Looks epic. Good job too @pixelastic Really excited to see this styled and working. Very cool feature!

**Comment by AdrieanKhisbe** [comment]
Voilà!  Styling is Ok.
Now it's the styling of the form that has to be tweak. 
(think I migh copy paste eslint one.)

**Comment by AdrieanKhisbe** [comment]
Gonna apply this https://github.com/eslint/eslint.github.io/commit/a6082ceee6f837cffec1c3e13ce345dba1a1fbed 
from @pixelastic  ;)

**Comment by mcdonnelldean** [comment]
Added WIP so I don't merge by mistake in one of my drive-through updates :D

**Comment by AdrieanKhisbe** [comment]
nopro. 
One last tweek to go and we are live ;)

**Comment by AdrieanKhisbe** [comment]
heheheh. Good to go! :bulb: 

@mcdonnelldean do a `npm run docs`, and see the magic :sparkles:

**Comment by mcdonnelldean** [comment]
Just pulled this down, I don't see the search box?

**Comment by AdrieanKhisbe** [comment]
you need either big or small screen. 
(middle with button would have require to much tweek, and would be time wasted since redising is n progress: cf https://github.com/AdrieanKhisbe/senecajs.org/pull/1 :) )

**Comment by mcdonnelldean** [comment]
Ah let me try again

Kindest Regards,

Dean

**Comment by mcdonnelldean** [comment]
Ah I see it now! This is epic! The orange is horrible but I'll be getting that fixed :D

Amazing work, can you list all involved in the contributors section and in comment so I can thank them on twitter please! 

Very, VERY impressive

**Comment by AdrieanKhisbe** [comment]
:wink:
Tim Carry  https://twitter.com/pixelastic and the amazing algolia team https://twitter.com/algolia
:fr: :mag: :bookmark_tabs: 
(already mentionned in commits and comment ;))

**Comment by mcdonnelldean** [comment]
Super stuff! https://twitter.com/senecajs/status/694159069800693760

Make sure @pixelastic is added to the contributor page. 

Huge thanks to everyone involved.
